src/components/form/form.types.ts

```diff
@@ -320,7 +320,128 @@ export enum FormLayoutType {
 }
 
 /**
- * @public
  * Represents the JSON schema with Lime specific options
+ * @public
+ */
+export interface FormSchema<T extends Record<string, any> = any>
+    extends JSONSchema7 {
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.4.1
+     */
+    items?: FormSchemaArrayItem<T> | Array<FormSchemaArrayItem<T>>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.4.2
+     */
+    additionalItems?: FormSchemaArrayItem<T>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.4.6
+     */
+    contains?: FormSchemaArrayItem<T>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.5.3
+     */
+    required?: Array<ReplaceObjectType<T, Extract<keyof T, string>, string>>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.5.4
+     */
+    properties?: ReplaceObjectType<
+        T,
+        FormSubKeySchema<T>,
+        Record<string, FormSchema>
+    >;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.7.1
+     */
+    allOf?: Array<FormSchemaArrayItem<T>>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.7.2
+     */
+    anyOf?: Array<FormSchemaArrayItem<T>>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.7.3
+     */
+    oneOf?: Array<FormSchemaArrayItem<T>>;
+
+    /**
+     * @see https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-9
+     */
+    $defs?: Record<string, FormSchema>;
```

Comment:
`$defs` and `definitions` (on line 420) link to the same section (where `definitions` is mentioned, but `$defs` is not). Is this correct? Do we need to keep both? If we need to keep both:

- Should we mention that they are the same thing, or what the differences are?
- We could perhaps place them next to each other, so it becomes a little easier to discover that there are two of them… 🙂

Reply:
Oh, don't do anything right now. I'm making changes here. Let me push first! 😅

Reply:
Nice, fixed. Not sure if it should be clarified even more or not.

Reply:
Looks good 👍

I don't think it needs further clarification. I guess it's whether `definitions` should be marked as deprecated perhaps? 🤷‍♂️